tabpfn_client/prompt_agent.py

```diff
@@ -100,18 +100,35 @@ def prompt_and_set_token(cls, user_auth_handler: "UserAuthenticationClient"):
                         )
                     )
             additional_info = cls.prompt_add_user_information()
-            is_created, message = user_auth_handler.set_token_by_registration(
-                email, password, password_confirm, validation_link, additional_info
+            is_created, message, access_token = (
+                user_auth_handler.set_token_by_registration(
+                    email, password, password_confirm, validation_link, additional_info
+                )
             )
             if not is_created:
                 raise RuntimeError("User registration failed: " + str(message) + "\n")
 
             print(
                 cls.indent(
-                    "Account created successfully! To start using TabPFN please click on the link in the verification email we sent you."
+                    "Account created successfully! To start using TabPFN please enter the secret key in the verification email we sent you."
                 )
                 + "\n"
             )
+            # verify token from email
+            verified = False
+            while not verified:
+                token = input(cls.indent("Verification Token: "))
+                verified, message = user_auth_handler.verify_email(token)
+                if not verified:
+                    print("\n" + cls.indent(str(message) + "Please try again!") + "\n")
+
+            print(
+                cls.indent(
+                    "Thank you for verifying your email successfully! Your access token is: "
+                )
+                + access_token
```

Comment:
Don't show the access token here. Only write:

cls.indent(
                    "Thank you for verifying your email successfully!

Reply:
To access the token there should be a function get_token and set_token

Reply:
Sure.

Reply:
Access token thing is in another PR. should we do all in one instead or keep it separate. usually separate notion boards have separate PRs